### PR TITLE
minor improvements to version rule unit tests

### PR DESCRIPTION
**Pull request by carlydf**
## What changed?
Replace `assert.Equal(t, expectedError, err)` with `assert.ErrorIs(...)` and `assert.ErrorContains(...)`.
Remove redundant `assert.Error(...)`

### CODE REVIEW

File: service/matching/version_rule_test.go [hunk 1]
```diff
@@ -266,8 +266,7 @@ func TestInsertAssignmentRuleMaxRules(t *testing.T) {
 
 	// insert fourth --> error
 	_, err = insertAssignmentRule(mkAssignmentRule("1", nil), data, clock, 0, maxRules)
-	assert.Error(t, err)
-	assert.Equal(t, errExceedsMaxAssignmentRules(4, maxRules), err)
+	assert.ErrorContains(t, err, errExceedsMaxAssignmentRules(4, maxRules).Error())
```
Comment (bergundy) [inline service/matching/version_rule_test.go:269]: nit: this doesn't check that the error is the same type but that's probably fine.

Comment (carlydf) [reply service/matching/version_rule_test.go:269]: yeahh it's not my fave but I'm going to leave it, since ErrorIs didnt work

**Comment by stephanos** [comment]
@carlydf let's merge it?

**Comment by yiminc** [comment]
No longer needed.

**Pull request closed without merging**